src/WinUsb/WinUsb.cs

```diff
@@ -0,0 +1,167 @@
+        /// A <paramref name="pipeIndex"/> value of 0 signifies the first endpoint that is associated with the interface, a value of 1 signifies the second endpoint,
+        /// and so on. <paramref name="pipeIndex"/> must be less than the value in the bNumEndpoints field of the interface descriptor.
+        /// </param>
+        /// <param name="pipeInformation">
+        /// A pointer, on output, to a caller-allocated <see cref="WINUSB_PIPE_INFORMATION"/> structure that contains pipe information.
+        /// </param>
+        /// <returns>
+        /// Returns <see langword="true"/> if the operation succeeds. Otherwise, this routine returns <see langword="false"/>,
+        /// and the caller can retrieve the logged error by calling <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        [DllImport(nameof(WinUsb), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool WinUsb_QueryPipe(SafeUsbHandle interfaceHandle, byte alternateInterfaceNumber, byte pipeIndex, out WINUSB_PIPE_INFORMATION pipeInformation);
+
+        /// <summary>
+        /// Writes data to a pipe.
+        /// </summary>
+        /// <param name="interfaceHandle">
+        /// An opaque handle to the interface that contains the endpoint with which the pipe is associated.
+        /// </param>
+        /// <param name="pipeID">
+        /// PipeID corresponds to the bEndpointAddress field in the endpoint descriptor.
+        /// </param>
+        /// <param name="buffer">
+        /// A caller-allocated buffer that contains the data to write.
+        /// </param>
+        /// <param name="bufferLength">
+        /// The number of bytes to write. This number must be less than or equal to the size, in bytes, of <paramref name="buffer"/>.
+        /// </param>
+        /// <param name="lengthTransferred">
+        /// A pointer to a ULONG variable that receives the actual number of bytes that were written to the pipe.
+        /// </param>
+        /// <param name="overlapped">
+        /// An optional pointer to an <see cref="NativeOverlapped"/> structure, which is used for asynchronous operations.
+        /// If this parameter is specified, <see cref="WinUsb_WritePipe"/> immediately returns, and the event is signaled when the operation is complete.
+        /// </param>
+        /// <returns>
+        /// Returns <see langword="true"/> if the operation succeeds. Otherwise, this routine returns <see langword="false"/>,
+        /// and the caller can retrieve the logged error by calling <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        [DllImport(nameof(WinUsb), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static unsafe extern bool WinUsb_WritePipe(SafeUsbHandle interfaceHandle, byte pipeID, void* buffer, int bufferLength, out int lengthTransferred, NativeOverlapped* overlapped);
```

Comment:
* I think declaring the `buffer` parameter as `byte*` would make more sense. Also adding `[Friendly(FriendlyFlags.In | FriendlyFlags.Array)]` to it would add an overload for passing in a `byte[]` which would be great.
* The `overlapped` parameter should be using `[Friendly(FriendlyFlags.Bidirectional | FriendlyFlags.Optional)]` (assuming that inputs have to be initialized on this struct, otherwise use `Out` instead of `Bidirectional`).